car_data.rb

```diff
@@ -0,0 +1,92 @@
+class Car
+
+  attr_reader :name, :gas_mileage, :n_cylinders, :horsepower, :weight, :quarter_mile_time, :salesman_name
+
+  def initialize(name, gas_mileage, n_cylinders, horsepower, weight, quarter_mile_time)
+
+    @name = name
+    @gas_mileage = gas_mileage 
+    @n_cylinders = n_cylinders
+    @horsepower = horsepower
+    @weight = weight*1000
+    @quarter_mile_time = quarter_mile_time
+    @salesman_name = "the "
+
+    @salesman_name += "powerful " if horsepower > 200
+
+    @salesman_name += "fuel efficient " if gas_mileage > 25
+
+    @salesman_name += name #does it matter if we have @name or self.name here?
+
+    def land_yacht?
+      self.weight > 3500
+    end
+
+    def to_s
+       "#{salesman_name}" #does it matter if we have self.salesman_name or @salesman_name here?
```

Comment:
Either of those is fine. In fact, since the return value of salesman_name is a string, you can drop the interpolation too.